StatementParser changes (based on my ABNAMRO file)

Johnny-Blue opened the pull request:
- programmersdigest.MT940Parser.Parser no need to implement IDisposable; using for the reader takes care of it

- StatementParser changes (based on my ABNAMRO file)
 :20: can be the first line in the file
 :28: added in ReadAccountIdentification; my file doesn't have a :28C:
 Statement can end w/o '-'; In my case it was only one statement in the file and ended with EOF

- Overriden some ToString() methods (Balance, StatementLine, Information)

- Added a simple sample app

- switched to .net core 3.0

luadag committed:
- switched to .net core 3.0

- programmersdigest.MT940Parser.Parser no need to implement IDisposable; using for the reader takes care of it

- StatementParser changes (based on my ABNAMRO file)
 :20: can be the first line in the file
 :28: added in ReadAccountIdentification; my file doesn't have a :28C:
 Statement can end w/o '-'; In my case it was only one statement in the file and ended with EOF

- Overriden some ToString() methods (Balance, StatementLine, Information)

- Added a simple sample app

programmersdigest commented:
Hey Johnny-Blue,

first of all, thanks for your work. Overall, I like the changes you have made and would like to merge most of them. The exception being the switch to .NET Core 3.0: I don't really see a benefit of targetting .NET Core 3.0, however users working with .NET Framework or older .NET Core versions would not be able to upgrade.

As such, I would much prefer if you could split your changes into different pull requests covering the five different topics you have named. This would make merging each change much easier and give an easier to understand commit history.

As it stands, I will close this pull request without merging, but hope you can split it up.

Johnny-Blue commented:
Hi,

Thanks for your reply. Not easy thing to do what you asked, because I added more changes to my version and I only have one branch (added some interfaces so I can inject it into my project and refactored it a bit). I'll do my best to accommodate your request though. Would it be okay to split it in two? Parser changes and everything else... let me know.

Since you're not fond of targeting .netcore30 how about targeting .netstandard2.1? This should solve the compatibility issues. I've changed your project to .netstandard2.1 and kept my added projects to .netcore3 (that's what I need, but it shouldn't impact in any way the people who want only the parser).

If you'd like you can preview my changes here:
https://github.com/Johnny-Blue/MT940Parser/tree/core30

Cheers,
